qt/mainwindow.cpp

```diff
@@ -54,14 +59,70 @@
 #endif // NO_DOWNLOADER
 
 
+namespace
+{
+// TODO(mgsergio): Consider getting rid of this class: just put everything
+// in TrafficMode.
+class TrafficDrawerDelegate : public TrafficDrawerDelegateBase
+{
+public:
+  explicit TrafficDrawerDelegate(qt::DrawWidget & drawWidget)
+    : m_framework(drawWidget.GetFramework())
+    , m_drapeApi(m_framework.GetDrapeApi())
+  {
+  }
+
+  void SetViewportCenter(m2::PointD const & center) override
+  {
+    m_framework.SetViewportCenter(center);
+  }
+
+  void DrawDecodedSegments(DecodedSample const & sample, int const sampleIndex) override
+  {
+    CHECK(!sample.GetItems().empty(), ("Sample must not be empty."));
+    auto const & points = sample.GetPoints(sampleIndex);
+
+    LOG(LINFO, ("Decoded segment", points));
+    m_drapeApi.AddLine(NextLineId(),
+                       df::DrapeApiLineData(points, dp::Color(0, 0, 255, 255))
+                       .Width(3.0f).ShowPoints(true /* markPoints */));
+  }
+
+  void DrawEncodedSegment(openlr::LinearSegment const & segment) override
+  {
+    auto const & points = segment.GetMercatorPoints();
+
+    LOG(LINFO, ("Encoded segment", points));
+    m_drapeApi.AddLine(NextLineId(),
+                       df::DrapeApiLineData(points, dp::Color(255, 0, 0, 255))
+                       .Width(3.0f).ShowPoints(true /* markPoints */));
+  }
+
+  void Clear() override
+  {
+    m_drapeApi.Clear();
+  }
+
+private:
+  string NextLineId() { return strings::to_string(m_lineId++); }
+
+  uint32_t m_lineId = 0;
```

Comment:
uint32_t is more than enough in this context. I can change to uint64_t if you feel like so.